### PR TITLE
[OSD-7738] Document all namespaces that are managed by Add-Ons

### PR DESCRIPTION
**Pull request by aliceh**
Added a script that finds all namespaces for managed-tenants addons and outputs them as a yaml document

### CODE REVIEW

File: resources/addons-namespaces/main.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ocs-converged-dev: openshift-storage
```
Comment (dustman9000) [inline resources/addons-namespaces/main.yaml:1]: To valid yaml, there needs to be a root element:
```suggestion
---
addon-namespaces:
   ocs-converged-dev: openshift-storage
```

File: scripts/addons-namespaces/generate-add-ons-list.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    parser.add_argument(
+        "--path",
+        "-p",
+        required=True,
+        help="The path to managed-tenants/addons/ directory [required]",
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
+        help="The destination of the output yaml file. Default managed-cluster-config/resources/addons-namespaces/main.yaml ",
```
Comment (dustman9000) [inline scripts/addons-namespaces/generate-add-ons-list.py:26]: No need to indicate the default in the help message. Just add a default value here.

File: scripts/addons-namespaces/generate-add-ons-list.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    )
+    parser.add_argument(
+        "--path",
+        "-p",
+        required=True,
+        help="The path to managed-tenants/addons/ directory [required]",
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
```
Comment (dustman9000) [inline scripts/addons-namespaces/generate-add-ons-list.py:25]:
```suggestion
        "-o",
        default="managed-cluster-config/resources/addons-namespaces/main.yaml",
```

File: scripts/addons-namespaces/generate-add-ons-list.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                namespace = ""
+                if file.endswith('addon.yaml'):
+                    file_to_search = open(file,"r")
+                    for line in file_to_search.readlines():
+                        if "targetNamespace" in line:
+                            ns = line.split(":")[1].strip()
+                            if namespace != ns:
+                                namespace = ns
+                                namescpace_dict[operator] = namespace 
+    if arguments.output == None :
```
Comment (dustman9000) [inline scripts/addons-namespaces/generate-add-ons-list.py:51]: No need for the check, just use the arguments.output value if you put the default value in the argument definition.

File: scripts/addons-namespaces/generate-add-ons-list.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        output="managed-cluster-config/resources/addons-namespaces/main.yaml"
+    else:
+        output=arguments.output
+    with open(output, 'w') as file:
+        yaml.dump(namescpace_dict, file )
```
Comment (dustman9000) [inline scripts/addons-namespaces/generate-add-ons-list.py:56]: Add extralines to separate functions.
```suggestion
        yaml.dump(namescpace_dict, file )
        
        
```

File: resources/addons-namespaces/main.yaml (hunk 1 above)
Comment (aliceh) [reply resources/addons-namespaces/main.yaml:1]: I fixed the script to generate a nested dict, so the yaml it generates has a root element.

**Pull request merged**